lib/stagehand/connection_adapter_extensions.rb

```diff
@@ -65,4 +65,8 @@ def handle_readonly_writes!
   class UnsyncedProductionWrite < StandardError; end
 end
 
-ActiveRecord::Base.connection.class.prepend(Stagehand::Connection::AdapterExtensions)
+begin
```

Comment:
If we found a better way to add these extensions that didn't rely on asking the connection instance what class it is, then we wouldn't need to have a connection at this point and could avoid the explosion.

Reply:
I am not sure I understand your comment. Are you saying that Stagehand should detect the reason why there is a database connection and only add its AdapterExtension in certain cases?

Is this even possible? Are there different types of `ActiveRecord::Base.connection`s?

Reply:
Yeah, I think it does it this way because you may be connected to a different database type, e.g. the one we use is for MySQL. Don't quote me on this, but basically all we need is for these methods to be used when called for the current database adapter class. So if we don't need to actually ask the `connection` object for what its superclass is, maybe because we're able to figure out how its selected by Rails in the first place, then we can just prepend the methods to that class.